packages/core/src/Cardano/types/Address.ts

```diff
@@ -24,29 +24,28 @@ const isRewardAccount = (address: string) => {
   }
 };
 
+/**
+ * Transform {@link value} into `Cardano.Address`,
+ * Resulting Address will be base58 in case of Byron era or bech32 in case of Shelley era or newer.
+ *
+ * @param value bech32 string, base58 string or hex-encoded bytes address.
+ * @throws {InvalidStringError} if value is invalid
+ */
+
 export const Address = (value: string): Address => {
   if (isAddress(value) && !isRewardAccount(value)) {
     return value as unknown as Address;
```

Comment:
Could early exit to not try to parse hex bytes from reward account, as well as give a better error message here
```ts
  if (isAddress(value)) {
    if (isRewardAccount(value)) {
      throw new InvalidStringError(`Address type can only be used for payment addresses`);
    }
    return value as unknown as Address;
  }
```

Reply:
[78c38c6](https://github.com/input-output-hk/cardano-js-sdk/pull/564/commits/78c38c6e4c75be7137e3718273e27d560c5e9dce). Added specific errors for:
1. non payment address
2. format not accepted
3. invalid address